### PR TITLE
add sample for lambda extension to kinesis data stream

### PR DESCRIPTION
**Pull request by ravinaik1312**
*Issue #, if available: * #58

*Description of changes:* The extension explains the overall approach to streamline and centralize log collection using Kinesis Data Streams. The extension runs a local HTTP listener and subscribes to a stream of function and platform logs using the Logs API. It buffers them and sends them to Kinesis Data Streams periodically. The demo gets deployed using AWS SAM. This contribution was possible because of @hariohmprasath 's awesome previous work( for Kinesis Firehose) which I could use as a reference and only change the parts pertaining to Kinesis Data Streams. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kinesis-stream-logs-extension-demo/template.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      - !Ref KinesisStreamLogsApiExtensionLayer
+      Environment:
+        Variables:
+          AWS_KINESIS_STREAM_NAME: !Ref KinesisStreamName
+      Policies:
+        - Statement:
+          - Sid: KinesisStreamFullAccess
+            Effect: Allow
+            Action:
+              - kinesis:*
```
Comment (julianwood) [inline kinesis-stream-logs-extension-demo/template.yaml]: Can this be scoped down a bit to just put records? Could this allow the Lambda function to delete the stream for example?

**Review by julianwood — COMMENTED** [review]
One suggestion added for additional function security in SAM template.

File: kinesis-stream-logs-extension-demo/template.yaml (hunk 1 above)
Comment (ravinaik1312) [reply kinesis-stream-logs-extension-demo/template.yaml]: Good call out, I'll get the permissions narrowed down.

**Review by julianwood — APPROVED** [review]

**Comment by julianwood** [comment]
Thanks for the contribution.

**Pull request merged**